pytorch_lightning/plugins/training_type/sharded_spawn.py

```diff
@@ -119,7 +119,7 @@ def new_process(self, trainer: "pl.Trainer", mp_queue: SimpleQueue) -> None:
         # Ensure that the scaler points to the correct process group
         # which is re-initialized in a new process
         if isinstance(self.precision_plugin, ShardedNativeMixedPrecisionPlugin):
-            self.precision_plugin.scaler = ShardedGradScaler()
+            self._precision_plugin.scaler = ShardedGradScaler()
```

Comment:
```suggestion
            self._precision_plugin._scaler = ShardedGradScaler()
```

Missing a setter: https://dev.azure.com/PytorchLightning/pytorch-lightning/_build/results?buildId=43066&view=logs&j=3afc50db-e620-5b81-6016-870a6976ad29&t=85e03a43-2268-568b-6f18-441ae2e4d13b